034-kafka-topic-replication-factor.md

```diff
@@ -0,0 +1,76 @@
+# Support Replication Factor Change for Kafka Topics
+
+Apache Kafka introduced the Replication Factor (RF) as the configuration that determines the number replicas kafka brokers will retain for a single data partition. A bigger number of the RF means both higher reliability for the data partition and higher infrastructure cost for topics.
+
+Apache Kafka allows users to change the RF for any topic partition through the native scripts during the runtime. However, it's not a lightweight process, because users usually need to decide which brokers to host new replicas and which replicas need to be dropped, or even conduct a cluster rebalance after the change of the RF.  
+
+The goal of this proposal is to offer users a simple approach to change the topic RF through the KafkaTopic Custom Resources and Cruise Control, in the cloud native, strimzi style. 
+
+## Current situation
+
+Currently, users have two options to change the topic RF:
+* Leverage the Native Kafka script to change the RF. Users need to generate the partition replica assignment and balance the kafka clusters manually if needed. 
+* Delete the topic and recreate the topic with the desired RF.
+
+Clearly none of the above approaches is user-friendly.
+
+When users attempt to change the RF in the KafkaTopic Resources, the Status of the KafkaTopic will become `NotReady` with `ReplicationFactorChangeException`. The following error message is shared with users.
+
+`Changing 'spec.replicas' is not supported. This KafkaTopic's 'spec.replicas' should be reverted to 4 and then the replication should be changed directly in Kafka.`
+
+## Motivation
+
+With this new feature, it will become much easier to achieve the desired tradeoff between the cost and data reliability in the kafka clusters. 
+
+For instance, users can lower the RFs for selective less critical topics to reduce the infrastructure cost (Compute, Storage and Network). Also, for some critical kafka topics, users can increase the data reliability through the bump of the RFs ONLY for these specific topics. 
+
+More importantly, users only need to specify the new RF number in the `KafkaTopic` resource and approve the change, in order to change the RF for kafka topics. The heavy lifting work, such as new replica placement, extra replica deletion and possible cluster rebalance, is completely taken care by the strimzi. 
+
+## Proposal
+
+### API
+The Kafka Topic RF can be changed by users in the following steps:
+* Update the `spec.replicas` properties in the target `KafkaTopic` Resource in the kubernetes cluster
+* Watch the `Status` field of the `KafkaTopic Resource`, for the status update of this RF change.
```

Comment:
I haven't tried the interaction with CC in order to change a RF but does it work as a rebalance? If yes, it means it's going to generate a proposal and then the user has to accept or refuse it before seeing the replication factor being increased. If confirmed, I don't think it's going to be just changing `spec.replicas` and watch the `Status` field. It would need a more "complex" state machine as we have for rebalancing.

Reply:
Based on the discussion, we agreed that we don't need to leverage the state machine to control this process, because the change of RF for a single Kafka topic is a very light weight process, compared with the cluster rebalance process.